test/health_failures_test_case.py

```diff
@@ -0,0 +1,120 @@
+from contextlib import contextmanager
+import os
+import json
+from unittest import mock
+from uuid import uuid4
+
+import boto3
+from mock import (
+    MagicMock,
+    patch,
+)
+from more_itertools import chunked
+from moto import (
+    mock_sts,
+    mock_sqs,
+    mock_dynamodb2,
+)
+import requests
+
+from app_test_case import LocalAppTestCase
+from azul import config
+from azul.health import HealthController
+from azul.json_freeze import (
+    freeze,
+    sort_frozen,
+)
+from azul.modules import load_app_module
+from retorts import ResponsesHelper
+
+
+class TestHealthFailures(LocalAppTestCase):
+
+    @classmethod
+    def lambda_name(cls) -> str:
+        return 'service'
+
+    @contextmanager
+    def _make_database(self):
+        database = boto3.resource('dynamodb', region_name='us-east-1')
+        table_settings = {
+            'TableName': config.dynamo_failure_message_table_name,
+            'KeySchema': [
+                {
+                    'AttributeName': 'MessageType',
+                    'KeyType': 'HASH'
+                },
+                {
+                    'AttributeName': 'SentTimeMessageId',
+                    'KeyType': 'RANGE'
+                }
+            ],
+            'AttributeDefinitions': [
+                {
+                    'AttributeName': 'MessageType',
+                    'AttributeType': 'S'
+                },
+                {
+                    'AttributeName': 'SentTimeMessageId',
+                    'AttributeType': 'S'
+                }
+            ],
+            'ProvisionedThroughput': {
+                'ReadCapacityUnits': 5,
+                'WriteCapacityUnits': 5
+            }
+        }
+        database.create_table(**table_settings)
+        with mock.patch.dict(os.environ, AWS_DEFAULT_REGION='us-east-1'):
+            try:
+                yield
+            finally:
+                table = database.Table(config.dynamo_failure_message_table_name)
+                table.delete()
+
+    @mock_sts
+    @mock_sqs
+    @mock_dynamodb2
+    def test_failures_endpoint(self):
+        indexer_app = load_app_module('indexer')
+        sqs = boto3.resource('sqs', region_name='us-east-1')
+        sqs.create_queue(QueueName=config.fail_queue_name)
+        fail_queue = sqs.get_queue_by_name(QueueName=config.fail_queue_name)
+        original_archive_fail_messages = HealthController.archive_fail_messages
+
+        def mocked_archive_fail_messages(self):
+            original_archive_fail_messages(self, 0)
+
+        with patch.object(HealthController, 'archive_fail_messages', new=mocked_archive_fail_messages):
```

Comment:
Unless I'm completely delirious, I think I mentioned in our conversation a couple days ago that the wait time should be extracted into a class attribute, not a method parameter. That's much easier to patch.

Reply:
You said I should parameterize `WaitTimeSeconds`, thats why I extracted it as a parameter.